File: easy/1. Two Sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32, i32> = HashMap::new();

        for (i, &val) in nums.iter().enumerate() {
            let need = target - val;

            if let Some(&idx) = map.get(&need) {
                return vec![idx, i as i32];
            }

            map.insert(val, i as i32);
        }

        vec![]
    }
}

File: easy/1. Two Sum/solution.java
class Solution{
    public int[] twoSum(int[] nums,int target){
        java.util.HashMap<Integer,Integer> m=new java.util.HashMap<>();
        for(int i=0;i<nums.length;i++){
            int c=target-nums[i];
            if(m.containsKey(c))return new int[]{m.get(c),i};
            m.put(nums[i],i);
        }
        return new int[0];
    }
}